common/beerocks/bcl/source/beerocks_utils.cpp

```diff
@@ -215,7 +215,7 @@ std::string utils::convert_channel_ext_above_to_string(bool channel_ext_above_se
 void utils::merge_list(std::vector<uint8_t> &ret, std::vector<uint8_t> &list)
 {
     std::vector<uint8_t>::iterator it;
-    for (it = list.begin(); it != list.end(); it++) {
+    for (it = list.begin(); it != list.end(); ++it) {
```

Comment:
For this one, a better fix is to remove the function entirely since it's not used.

Or failing that, use ranged loop:
```
    for (element : list) {
```

Reply:
@RanRegev , I prefer @arnout 's approach, not only because it is easier, but also because this function is too bad to keep:
* it is unused
* use case too niche to happen often
* code style does not correspond to current guidelines
* the name is misleading (should be something like `merge_non_zero_values_from_list`)